core/domain/story_domain.py

```diff
@@ -341,6 +348,11 @@ def validate(self):
             node.validate()
             if node.id == self.starting_node_id:
                 initial_node_is_present = True
+            # Checks whether the number in the id of any node is greater than
+            # the value of next_node_id.
+            if int(node.id.replace('node_', '')) >= self.next_node_id:
+                raise utils.ValidationError(
+                    'The node with id %s is out of bounds.' % node.id)
             node_id_list.append(node.id)
 
         if not initial_node_is_present:
```

Comment:
Where's the validation for next_node_id?

Reply:
Done, added validations.